yb-voyager/cmd/importData.go

```diff
@@ -355,15 +353,38 @@ func importData() {
 		fmt.Printf("Number of parallel imports jobs at a time: %d\n", parallelism)
 	}
 
-	if parallelism > SPLIT_FILE_CHANNEL_SIZE {
-		splitFileChannelSize = parallelism + 1
+	determineTablesToImport()
+	if startClean {
+		cleanImportState()
+	}
+	if target.VerboseMode {
+		fmt.Printf("all the tables to be imported: %v\n", allTables)
+		fmt.Printf("tables left to import: %v\n", importTables)
 	}
-	splitFilesChannel := make(chan *SplitFileImportTask, splitFileChannelSize)
 
-	generateSmallerSplits(splitFilesChannel)
-	go doImport(splitFilesChannel, parallelism, connPool)
-	checkForDone()
-	time.Sleep(time.Second * 2)
+	if len(importTables) == 0 {
+		fmt.Printf("All the tables are already imported, nothing left to import\n")
+	} else {
+		fmt.Printf("Importing tables: %v\n", importTables)
+		initializeImportDataStatus(exportDir, importTables)
+		if !disablePb {
+			go importDataStatus()
+		}
+		poolSize := 20
+		if parallelism > poolSize {
+			poolSize = parallelism + parallelism/2
+		}
+		// The code can produce `poolSize` number of batches at a time. But, it can consume only
+		// `parallelism` number of batches at a time.
+		batchImportPool = pool.New().WithMaxGoroutines(poolSize)
```

Comment:
With this change, we are relying on `connPool` to control the parallelism.

Where the connection fetching is being done within the `connPool.WithConn(copyFn)`.
but before all this in `doOneImport()` function we are doing all renaming from `*.C` to `*.P` and **even though max `*.P` split count should be equal to parallelism but they will be equal to poolSize here**

This is not a user facing problem but for developers while debugging this might be confusing.

Reply:
This is on us to understand it once. For now, let's go with what is implemented.